src/multimeditron/model/model.py

```diff
@@ -14,6 +14,154 @@
 
 logger = logging.getLogger(__name__)
 
+@dataclass
+class ChatTemplate:
+    """
+    A generic chat template class to serialize conversation messages
+    for different LLM families (LLaMA, Qwen, Apertus, etc.).
+    """
+    name: str = "custom"
+    roles: Dict[str, str] = field(default_factory=lambda: {
+        "user": "user",
+        "assistant": "assistant",
+        "system": "system"
+    })
+    bos_token: Optional[str] = None
+    eos_token: Optional[str] = None
+    stop_tokens: List[str] = field(default_factory=list)
+
+    # Explicit delimiters for each message type
+    delimiters: Dict[str, Dict[str, str]] = field(default_factory=dict)
+
+    # Optional serialization override
+    serialize_fn: Optional[Callable[[List[Dict[str, str]]], str]] = None
+
+    # ================================================================
+
+    def format(self, messages: List[Dict[str, str]]) -> str:
+        """Serialize a list of messages into the model’s expected chat format."""
+        if self.serialize_fn:
+            return self.serialize_fn(messages)
+
+        # Default: apply delimiters per role
+        text = []
+        if self.bos_token:
+            text.append(self.bos_token)
+
+        for msg in messages:
+            role = msg.get("role", "user")
+            content = msg.get("content", "")
+            d = self.delimiters.get(role, {})
+            start, end = d.get("start", ""), d.get("end", "")
+            text.append(f"{start}{content}{end}")
+
+        if self.eos_token:
+            text.append(self.eos_token)
+        return "\n".join(text)
+
+    # ================================================================
+    # Built-in templates
+    # ================================================================
+
+    @staticmethod
+    def from_name(name: str) -> "ChatTemplate":
```

Comment:
Use future annotations, it's cleaner

```suggestion
    def from_name(name: str) -> ChatTemplate:
```